### PR TITLE
Cockpit 265

### PR DESCRIPTION
**Pull request by garrett**
Preview: https://garrett.github.io/cockpit-project.github.io/blog/cockpit-265.html

### CODE REVIEW

**Comment by skobyda** [comment]
I tried to record a better gif for animated rows but hmm... Is it just me or does it look of different color variation?

**Comment by martinpitt** [comment]
@skobyda The animation in the gif looks fine here. I tried it on the services page with `sudo systemd-run --unit aaaa --wait sleep 10` and it looks pretty much like your animation. Of course animated GIF is always much more choppy like a smooth "real" video, but it gets the point across just fine. Thanks!

File: _posts/2022-03-16-cockpit-265.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+title: Cockpit 265
+author: skobyda
+date: '2022-03-16'
+tags: cockpit, machines
+slug: cockpit-265
+category: release
+summary: 'Animated lists, x-forwarder-for support, crypto policies, disk serial number, metrics showing busiest CPU core, manifest overrides location'
```
Comment (martinpitt) [inline _posts/2022-03-16-cockpit-265.md:8]: Seems a little longish, but then again we don't actually show these summaries anywhere. But it's X-Forwarded-For, not lower-case and forwarde*r*

File: _posts/2022-03-16-cockpit-265.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+When new items appear on a page, they are highlighted for a short while to spot them more easily.
+
+![Peek 2022-03-15 07-47]({{ site.baseurl }}/images/265-animate-new-rows-in-lists.gif)
+
+## Support for `X-Forwarded-For`
+
+When run in a configuration involving a reverse proxy server, Cockpit can now report the client IP address as provided to it by the proxy server, rather than using the IP of the proxy server itself.  This impacts what appears in system login logs, for example, and may have an impact on authentication, depending on which pam modules are in use.
+
+Support for this feature must be explicitly enabled via the new `ForwardedForHeader` option in the configuration file.
```
Comment (martinpitt) [inline _posts/2022-03-16-cockpit-265.md:27]: Let's turn "configuration file" into a link to https://cockpit-project.org/guide/latest/cockpit.conf.5.html

File: _posts/2022-03-16-cockpit-265.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+[Manifest override files](https://cockpit-project.org/guide/latest/packages.html#package-replace) are a way to customize or hide Cockpit's page menu entries. Until now these needed to be put next to the `manifest.json` which they modify, for example `/usr/share/cockpit/systemd/override.json`. However, `/usr/share/` is package manager territory,  and admins should not (and in some cases even cannot) write files there. With this Cockpit version, these overrides can now be placed into `/etc/cockpit/systemd.override.json` for global customizations, or `~/.config/cockpit/systemd.override.json` for user specific ones.
+
+## Crypto policies support
+
+Cockpit now has basic support for displaying and changing a subset of crypto policies (default, future, legacy). The current configured crypto policy is shown in the "Configuration" card.
+
+![158211328-fd8936be-5127-4e30-9c4b-5ba251c7d60a]({{ site.baseurl }}/images/265-crypto-policies-support.png)
+
+Crypto policy changes can be applied directly or later, when later is selected the health card shows that a reboot is required to fully apply the changed crypto policy.
```
Comment (martinpitt) [inline _posts/2022-03-16-cockpit-265.md:39]:
```suggestion
Crypto policy changes can be applied immediately or later. When "Apply only" is selected, the Health card shows that a reboot is required to fully apply the changed crypto policy.
```

File: _posts/2022-03-16-cockpit-265.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## Metrics: Show busiest CPU core
+
+On multi-core systems, the Metrics page now shows an additional usage bar for the CPU core with the highest usage. This points out bottlenecks and performance problems which happen because only a few cores (or even a single one) get fully used, while many others are mostly idle. In that case, the average CPU usage will be low.
+
+![screenshot of show busiest cpu core]({{ site.baseurl }}/images/265-metrics-show-busiest-cpu-core.png)
+
+## Machines: Disk serial number
+
+Disks identifier can be specified during a disk attachment. Its use cases include:
```
Comment (martinpitt) [inline _posts/2022-03-16-cockpit-265.md:51]:
```suggestion
A disk identifier can be specified during attachment. Use cases include:
```

File: _posts/2022-03-16-cockpit-265.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+- User might have licensed functionality inside the VM tied to a specific
+disk serial number (most common with Windows guests)
+- Setting up multipath storage
+- This disk will be identifiable from inside the guest by /dev/disk/by-id.
```
Comment (martinpitt) [inline _posts/2022-03-16-cockpit-265.md:55]:
```suggestion
- This disk will be identifiable from inside the guest in `/dev/disk/by-id/`.
```

File: _posts/2022-03-16-cockpit-265.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+* [For your Linux system](https://cockpit-project.org/running.html)
+
+* [Cockpit Source Tarball](https://github.com/cockpit-project/cockpit/releases/tag/265)
+* [Cockpit Fedora 36](https://bodhi.fedoraproject.org/updates/FEDORA-2022-ad93e6cdcb)
+* [Cockpit Fedora 35](https://bodhi.fedoraproject.org/updates/FEDORA-2022-ec9612ac99)
+* [Cockpit Client](https://flathub.org/apps/details/org.cockpit_project.CockpitClient)
+* [cockpit-machines Source Tarball](https://github.com/cockpit-project/cockpit-machines/releases/tag/264)
+* [cockpit-machines Fedora 36](https://bodhi.fedoraproject.org/updates/FEDORA-2022-7e1cbde0f7)
+* [cockpit-machines Fedora 35](https://bodhi.fedoraproject.org/updates/FEDORA-2022-2cd2b4d57d)
+* [Cockpit Client](https://flathub.org/apps/details/org.cockpit_project.CockpitClient)
```
Comment (martinpitt) [inline _posts/2022-03-16-cockpit-265.md:73]: We didn't actually release a new Client, and none of the new features affect it -- i.e. we don't need to. Let's just drop the line.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks!

File: _posts/2022-03-16-cockpit-265.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+## Manifest overrides in /etc and ~/.config directories
+
+[Manifest override files](https://cockpit-project.org/guide/latest/packages.html#package-replace) are a way to customize or hide Cockpit's page menu entries. Until now these needed to be put next to the `manifest.json` which they modify, for example `/usr/share/cockpit/systemd/override.json`. However, `/usr/share/` is package manager territory,  and admins should not (and in some cases even cannot) write files there. With this Cockpit version, these overrides can now be placed into `/etc/cockpit/systemd.override.json` for global customizations, or `~/.config/cockpit/systemd.override.json` for user specific ones.
+
+## Crypto policies support
+
+Cockpit now has basic support for displaying and changing a subset of crypto policies (default, future, legacy). The current configured crypto policy is shown in the "Configuration" card.
+
+![158211328-fd8936be-5127-4e30-9c4b-5ba251c7d60a]({{ site.baseurl }}/images/265-crypto-policies-support.png)
```
Comment (martinpitt) [inline _posts/2022-03-16-cockpit-265.md:37]: This image is blurry, and it couldn't hurt to equalize the frame width a bit.

Also, please let's not use these UUIDs as alt names, but something readable like "crypto policies in Configuration card"

File: _posts/2022-03-16-cockpit-265.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+Crypto policy changes can be applied immediately or later. When "Apply only" is selected, the Health card shows that a reboot is required to fully apply the changed crypto policy.
+
+![158211149-36822bc4-a47e-4501-a01a-a82f7bbf3ec2]({{ site.baseurl }}/images/265-crypto-policies-support-2.png)
```
Comment (martinpitt) [inline _posts/2022-03-16-cockpit-265.md:41]: This is blurry as well, looks like being scaled up? Same UUID alt-name issue.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Some further suggestions, but this is coming together. Thanks!

This PR adds a lot of images which are not being used in the blog post. Can you please remove them?

@garrett or @allisonkarlitskaya , do you have time for a round of English cleanup?

**Comment by skobyda** [comment]
Looking at the size of the images, it seems they were artificially enlarged (the buttons also look bigger). For example [here](https://github.com/cockpit-project/cockpit/pull/16860), even on github the images look larger than natural

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]

**Comment by garrett** [comment]
Relevant for today:

github
(mod of https://xkcd.com/303/)

Meanwhile, it's back enough for me to review this, I think?

**Comment by skobyda** [comment]
Fixed @allisonkarlitskaya comments

**Comment by skobyda** [comment]
> Meanwhile, it's back enough for me to review this, I think?

Yeah, altough it seems that the build [here](https://garrett.github.io/cockpit-project.github.io/blog/cockpit-265.html) still has old screenshots

**Comment by garrett** [comment]
> Yeah, altough it seems that the build [here](https://garrett.github.io/cockpit-project.github.io/blog/cockpit-265.html) still has old screenshots

Does it? You might need to shift-reload.

**Comment by skobyda** [comment]
https://user-images.githubusercontent.com/42733240/158840254-3c2963a3-e7d3-475c-bb6e-ba27b8cf2a00.mp4

**Comment by skobyda** [comment]
Uploading out.mp4…

**Comment by skobyda** [comment]
https://user-images.githubusercontent.com/42733240/158847827-48903fd1-b5c2-4153-8f60-36c0b0810841.mp4

File: _posts/2022-03-16-cockpit-265.md (hunk 7 above)
Comment (martinpitt) [reply _posts/2022-03-16-cockpit-265.md:37]: Please still fix these horrible UUIDs.

**Review by martinpitt — COMMENTED** [review]
Thanks! The images look good now, just these UUIDs need to be replaced with something decent for humans (alt-text). Otherwise, consider this +1 from me; I'll leave the rest (verbiage and video fine-tuning) beween Garrett and you.

**Pull request merged**